### ISSUE TITLE
Memory Leak in syncProviderWrapper

### ISSUE THREAD

**Issue by khagerma** [virtual-kubelet/virtual-kubelet#1045]
TL;DR: The deletePods map in syncProviderWrapper grows without bounds.

In addition to this, the code which relies on deletePods only checks if the key exists, but the entire k8s Pod object is stored, amplifying the effects of this memory leak.

If it is not possible to automatically detect when pods are "gone forever", then pods should be removed from this list after some timeout (maybe 5 minutes?)

This issues was found using a combination of tools, but was made obvious when investigating a core dump with delve:

Note how this map has `1863`+ entries.  This was taken on a system which was running only ~20 pods running at once (maybe ~30 including k8s containers).
```
(dlv) p p.deletedPods
sync.Map {
	mu: sync.Mutex {state: 0, sema: 0},
	read: sync/atomic.Value {
		v: interface {}(sync.readOnly) *(*interface {})(0xc00010dd40),},
	dirty: map[interface {}]*sync.entry [
		*(*interface {})(0xc002614008): *(*sync.entry)(0xc000600f80), 
		*(*interface {})(0xc002614018): *(*sync.entry)(0xc00019b468), 
		*(*interface {})(0xc002614028): *(*sync.entry)(0xc00082a138), 
		*(*interface {})(0xc002614038): *(*sync.entry)(0xc00222a068), 
		*(*interface {})(0xc0026140d8): *(*sync.entry)(0xc00053cde8), 
		*(*interface {})(0xc0026140e8): *(*sync.entry)(0xc00000f0d8), 
		*(*interface {})(0xc0026140f8): *(*sync.entry)(0xc00019b180), 
		*(*interface {})(0xc0026141a8): *(*sync.entry)(0xc0007eab08), 
		*(*interface {})(0xc002614278): *(*sync.entry)(0xc0003f3080), 
		*(*interface {})(0xc002614288): *(*sync.entry)(0xc000a06c90), 
		*(*interface {})(0xc002614298): *(*sync.entry)(0xc0003f2c98), 
		*(*interface {})(0xc002614418): *(*sync.entry)(0xc00019a4c0), 
		*(*interface {})(0xc002614428): *(*sync.entry)(0xc000a07230), 
		*(*interface {})(0xc0026144e8): *(*sync.entry)(0xc0003f3200), 
		*(*interface {})(0xc0026145b8): *(*sync.entry)(0xc0016804f0), 
		*(*interface {})(0xc0026145c8): *(*sync.entry)(0xc000a063a8), 
		*(*interface {})(0xc0026145d8): *(*sync.entry)(0xc0003f2350), 
		*(*interface {})(0xc0026145e8): *(*sync.entry)(0xc00000f708), 
		*(*interface {})(0xc002614688): *(*sync.entry)(0xc00082a3f0), 
		*(*interface {})(0xc002614758): *(*sync.entry)(0xc000a06c68), 
		*(*interface {})(0xc002614768): *(*sync.entry)(0xc0003f2e08), 
		*(*interface {})(0xc002614778): *(*sync.entry)(0xc00019b830), 
		*(*interface {})(0xc002614828): *(*sync.entry)(0xc000aba310), 
		*(*interface {})(0xc002614838): *(*sync.entry)(0xc001680158), 
		*(*interface {})(0xc002614848): *(*sync.entry)(0xc000aba428), 
		*(*interface {})(0xc002614858): *(*sync.entry)(0xc000a06600), 
		*(*interface {})(0xc0026148f8): *(*sync.entry)(0xc0003f2938), 
		*(*interface {})(0xc002614908): *(*sync.entry)(0xc000a06e88), 
		*(*interface {})(0xc002614918): *(*sync.entry)(0xc00000f658), 
		*(*interface {})(0xc0026149c8): *(*sync.entry)(0xc0007eac88), 
		*(*interface {})(0xc0026149d8): *(*sync.entry)(0xc00000f7b0), 
		*(*interface {})(0xc0026149e8): *(*sync.entry)(0xc000a07718), 
		*(*interface {})(0xc0026149f8): *(*sync.entry)(0xc000aba178), 
		*(*interface {})(0xc002614a98): *(*sync.entry)(0xc00053c200), 
		*(*interface {})(0xc002614aa8): *(*sync.entry)(0xc00019b6b0), 
		*(*interface {})(0xc002614c38): *(*sync.entry)(0xc00019abb8), 
		*(*interface {})(0xc002614c48): *(*sync.entry)(0xc00082ad40), 
		*(*interface {})(0xc002614d08): *(*sync.entry)(0xc0003f2840), 
		*(*interface {})(0xc002614d18): *(*sync.entry)(0xc00053d408), 
		*(*interface {})(0xc002614d28): *(*sync.entry)(0xc00019b460), 
		*(*interface {})(0xc002614d38): *(*sync.entry)(0xc00053c830), 
		*(*interface {})(0xc002614d48): *(*sync.entry)(0xc00019b4e8), 
		*(*interface {})(0xc002614d58): *(*sync.entry)(0xc001680650), 
		*(*interface {})(0xc002614d68): *(*sync.entry)(0xc000aba988), 
		*(*interface {})(0xc002614d78): *(*sync.entry)(0xc00000f3d0), 
		*(*interface {})(0xc00262e0d8): *(*sync.entry)(0xc00053cf50), 
		*(*interface {})(0xc002614dd8): *(*sync.entry)(0xc0003f2b88), 
		*(*interface {})(0xc002614ea8): *(*sync.entry)(0xc000601178), 
		*(*interface {})(0xc002614eb8): *(*sync.entry)(0xc0006012a8), 
		*(*interface {})(0xc002614ec8): *(*sync.entry)(0xc000600b98), 
		*(*interface {})(0xc002614ed8): *(*sync.entry)(0xc00053d330), 
		*(*interface {})(0xc002614ee8): *(*sync.entry)(0xc001680368), 
		*(*interface {})(0xc002614f78): *(*sync.entry)(0xc0006007b0), 
		*(*interface {})(0xc002614f88): *(*sync.entry)(0xc0016801b0), 
		*(*interface {})(0xc002614f98): *(*sync.entry)(0xc001680bb0), 
		*(*interface {})(0xc002615048): *(*sync.entry)(0xc0003f2cc8), 
		*(*interface {})(0xc002615058): *(*sync.entry)(0xc000abaad0), 
		*(*interface {})(0xc002615118): *(*sync.entry)(0xc00053d250), 
		*(*interface {})(0xc002615128): *(*sync.entry)(0xc0003f2c10), 
		*(*interface {})(0xc002615138): *(*sync.entry)(0xc0003f2a68), 
		*(*interface {})(0xc002615148): *(*sync.entry)(0xc0003f2ec8), 
		*(*interface {})(0xc002615158): *(*sync.entry)(0xc000abaa68), 
		*(*interface {})(0xc0026151e8): *(*sync.entry)(0xc00053c4a8), 
		*(*interface {})(0xc0026151f8): *(*sync.entry)(0xc0006006e8), 
		...+1863 more
	],
	misses: 67,}
```
